modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpDownloader.java

```diff
@@ -323,6 +327,20 @@ private void cleanDatabases() {
         stats = stats.expiredDatabases((int) expiredDatabases);
     }
 
+    synchronized void deleteIndex() {
```

Comment:
Right beneath this method is the `onCancelled()` method, which runs on the node executing the task (on a GENERIC thread, so we should be fine). The task should not be scheduled again until after `markAsCompleted()` is called, so as long as we have a way to avoid running the delete at the same time as the update (synchronized is probably fine?) then the delete will happen always happen on the right node.